### ISSUE TITLE
Handle imgur links better

### ISSUE THREAD

**Issue by deepdesigns6** [travis-g/Voat-Enhancement-Suite#40]
These links fail to show the expand image feature/button:
http://i.imgur.com/5GcbdGi.png?1
http://imgur.com/7oRYwew
http://imgur.com/cMEySbM

**Comment by travis-g** [comment]
The first screenshot is from Reddit, and from the appearance of the user bar RES isn't enabled. Do you have an example of this happening between Voat and VES?

**Comment by deepdesigns6** [comment]
Those are imgur links that don't work, not screenshots of the issue.

**Comment by deepdesigns6** [comment]
To start, we can fetch all imgur links on the page and match them to the problematic URLs, such as ones without an extension. Pass it to the imgur api to get the link and replace the href attribute. Then run
UI.ExpandoManager.execute(); and it will add the expando icon.

**Comment by travis-g** [comment]
>Those are imgur links that don't work, not screenshots of the issue.

Ah, ok. And this issue can be seen to affect Voat itself (wIth VES disabled)?

**Comment by deepdesigns6** [comment]
Yes, this is an issue with Voat itself.

**Comment by travis-g** [comment]
It seems as though you've got an idea for how to fix the issue using Voat's code (which should not be referenced by VES). Have you submitted an issue in their repo?

**Comment by deepdesigns6** [comment]
I will, I just wanted to fix it for myself because it's very annoying. Also, RES handles imgur galleries different than the reddit source, so it may be something we want to look into.

I put up a new branch with a new module. It seems to work as you can pretty much just add .jpg to anything on imgur and it work. So, next step is to use the imgur api to get a proper link.

Anyway, might be nice to have as an option to turn on. If you go to /v/pics you'll see a ton of links that don't have expando enabled. I doubt they'll be getting a ton of software stuff done while they're dealing with the recent surge.

**Comment by travis-g** [comment]
To place a firm guideline for the future, the goal of VES is not to fix Voat bugs. 

That said, if you repackage your module to be a subset of a `voatHotfixes` module I guess VES will host some temporary fixes while Voat gets back on its feet.

**Comment by deepdesigns6** [comment]
I agree whole heartedly especially since fixes are dependent on markup which they could change at a moments notice. Otherwise, we'd be scraping every link on the page which is bad form. Even if we did that, where does it end?

Anyway, I'll move it to hotfixes for now. Is there a chacne you will be publishing a dev branch soon? I'm still working off of the almost two week old master.

**Comment by travis-g** [comment]
Dev will be updated within 8 hours.

**Comment by deepdesigns6** [comment]
The plot thickens. Voat handles images via extension, say a link that ends in .jpg. It does not have any service-specific handling (apparently). So, basically, a link submitted "incorrectly" is actually showing correctly. 

Do you want VES to tackle imgur, slimgur, and the other one I can't remember the name of? If so I can continue to hack away at it, quite literally.

**Comment by travis-g** [comment]
How deep have you gone in Voat's source to find this out? And is this on the client- or server-side of Voat?

If it's not implemented smartly point it out to *someone* developing Voat's source.

See what it'd take to fix the issue, and find the simplest, most efficient way to go about it. Hotfix it with VES and send the solution to Voat.

**Comment by deepdesigns6** [comment]
Looks like a pretty easy fix:

https://github.com/voat/voat/blob/cad35c1647e651c572c791353f44364288f482dc/Whoaverse/Whoaverse/Scripts/whoaverse.ui.js#L273

I'm down for hotfixing it and attempting to get voat to integrate . Looks like they integrated imgur albums and a few other services so I don't see why they wouldn't do this.

**Comment by travis-g** [comment]
> Dev will be updated within 8 hours.

Voat's been down again, so I pushed the probably-stable changes I have ready into `dev-stash`, but I haven't been able to test them yet. Ignore the settings menu as it doesn't have the CSS to go with it :laughing:

**Comment by travis-g** [comment]
> Dev will be updated within 8 hours.

Voat's been down again, so I pushed the probably-stable changes I have ready into `dev-stash`, but I haven't been able to test them yet. Ignore the settings menu as it doesn't have the CSS to go with it :laughing:

**Comment by deepdesigns6** [comment]
Back up for me finally.. I'm a bit busier today but I'll try to test it out.
